ddtrace/mocktracer/mockspan.go

```diff
@@ -7,282 +7,122 @@ package mocktracer // import "github.com/DataDog/dd-trace-go/v2/ddtrace/mocktrac
-		s.tags = make(map[string]interface{}, 1)
+	switch k {
+	case ext.SpanName:
+		return s.Name
+	case ext.ServiceName:
+		return s.Service
+	case ext.ResourceName:
+		return s.Resource
+	case ext.SpanType:
+		return s.Type
 	}
-	if key == ext.SamplingPriority {
-		switch p := value.(type) {
-		case int:
-			s.context.setSamplingPriority(p)
-		case float64:
-			s.context.setSamplingPriority(int(p))
+	if s.Meta != nil {
+		if r, ok := s.Meta[k]; ok {
+			return r
 		}
 	}
-	s.tags[key] = value
-}
-
-func (s *mockspan) FinishTime() time.Time {
-	s.RLock()
-	defer s.RUnlock()
-	return s.finishTime
-}
-
-func (s *mockspan) StartTime() time.Time { return s.startTime }
-
-func (s *mockspan) Tag(k string) interface{} {
-	s.RLock()
-	defer s.RUnlock()
-	return s.tags[k]
-}
-
-func (s *mockspan) Tags() map[string]interface{} {
-	s.RLock()
-	defer s.RUnlock()
-	// copy
-	cp := make(map[string]interface{}, len(s.tags))
-	for k, v := range s.tags {
-		cp[k] = v
+	if s.Metrics != nil {
+		if r, ok := s.Metrics[k]; ok {
+			return r
+		}
 	}
-	return cp
-}
-
-func (s *mockspan) TraceID() uint64 { return s.context.traceID }
-
-func (s *mockspan) SpanID() uint64 { return s.context.spanID }
-
-func (s *mockspan) ParentID() uint64 { return s.parentID }
-
-func (s *mockspan) OperationName() string {
-	s.RLock()
-	defer s.RUnlock()
-	return s.name
-}
-
-// SetOperationName resets the original operation name to the given one.
-func (s *mockspan) SetOperationName(operationName string) {
-	s.Lock()
-	defer s.Unlock()
-	s.name = operationName
-	return
-}
-
-// BaggageItem returns the baggage item with the given key.
-func (s *mockspan) BaggageItem(key string) string {
-	return s.context.baggageItem(key)
+	return nil
 }
 
-// SetBaggageItem sets a new baggage item at the given key. The baggage
-// item should propagate to all descendant spans, both in- and cross-process.
-func (s *mockspan) SetBaggageItem(key, val string) {
-	s.context.setBaggageItem(key, val)
-	return
-}
-
-// Finish finishes the current span with the given options.
-func (s *mockspan) Finish(opts ...ddtrace.FinishOption) {
-	var cfg ddtrace.FinishConfig
-	for _, fn := range opts {
-		fn(&cfg)
-	}
-	var t time.Time
-	if cfg.FinishTime.IsZero() {
-		t = time.Now()
-	} else {
-		t = cfg.FinishTime
-	}
-	if cfg.Error != nil {
-		s.SetTag(ext.Error, cfg.Error)
+func (s *Span) Tags() map[string]interface{} {
+	r := make(map[string]interface{})
+	if s == nil {
+		return r
 	}
-	if cfg.NoDebugStack {
-		s.SetTag(ext.ErrorStack, "<debug stack disabled>")
+	for k, v := range s.Meta {
+		r[k] = v
 	}
-	s.Lock()
-	defer s.Unlock()
-	if s.finished {
-		return
+	for k, v := range s.Metrics {
+		r[k] = v
 	}
-	s.finished = true
-	s.finishTime = t
-	s.tracer.addFinishedSpan(s)
+	r[ext.SpanName] = s.Name
+	r[ext.ServiceName] = s.Service
+	r[ext.ResourceName] = s.Resource
+	r[ext.SpanType] = s.Type
+	return r
 }
 
-// String implements fmt.Stringer.
-func (s *mockspan) String() string {
+func (s *Span) String() string {
```

Comment:
It's weird that the other methods appear to expect/deal with a `nil` receiver, but this one doesn't... I'd argue either all need to gracefully deal with a `nil` receiver, or none should.